crates/burn-import/src/onnx/dim_inference.rs

```diff
@@ -195,28 +195,36 @@ fn concat_update_outputs(node: &mut Node) {
 }
 
 fn reshape_update_outputs(node: &mut Node) {
-    assert_eq!(node.inputs.len(), 2);
-
-    let shape = if let Some(Data::Int64s(ref shape)) = node.inputs[1].value {
-        shape
+    let shape = if node.inputs.len() == 2 {
+        // get the values while making sure the types are correct
+        match &node.inputs[1].value {
+            Some(value) => match value {
+                Data::Int64s(shape) => Some(shape.clone()),
+                _ => panic!("Reshape: invalid input types"),
+            },
+            None => None,
+        }
     } else {
-        panic!("Reshape: int64s shape is expected per ONNX spec");
+        node.attrs
+            .iter()
+            .find_map(|(key, value)| match key.as_str() {
+                "shape" => Some(value.clone().into_i64s()),
+                _ => None,
+            })
```

Comment:
If we're just looking for a single specific attribute, couldn't we use `.get(attr)` instead?

Reply:
Done